File: src/main/java/rustleund/fightingfantasy/framework/Closure.java
package rustleund.fightingfantasy.framework;

public interface Closure {

	/**
	 * Execute a game task.
	 * 
	 * @param gameState The current {@link GameState}
	 * 
	 * @return <code>true</code> if the execution was successful, <code>false</code> otherwise
	 */
	boolean execute(GameState gameState);

}

File: src/main/java/rustleund/fightingfantasy/framework/closures/Closure.kt
package rustleund.fightingfantasy.framework.closures

import rustleund.fightingfantasy.framework.base.GameState

/**
 * @see [execute]
 */
fun interface Closure {

    /**
     * Execute a game task.
     *
     * @param gameState The current [GameState]
     *
     * @return `true` if the execution was successful, `false` otherwise
     */
    fun execute(gameState: GameState): Boolean
}